papas/simulation/Simulator.h

```diff
@@ -193,17 +191,10 @@ class Simulator {
    @param[in] parentid used to update history
    */
   void storeSmearedTrack(Track&& smearedtrack, Identifier parentid);
-  /**
-   Propagates a particle to the specified cyclinder
-   @param[in] cylinder the cylinder where  we want to know where the particle arrives
-   @param[in] ptc the particle to be propagated
-   */
-  void propagate(const SurfaceCylinder& cylinder, PFParticle& ptc);
-  /**
-   Propagates a particle to the ECAL and Hcal cyclinders
-   @param[in] ptc the particle to be propagated
+  /** Return shared ptr to the appropriate propagator (straightline or helix)
+   @param[in] charge charge of particle
    */
-  void propagateAllLayers(PFParticle& ptc);
+  std::shared_ptr<Propagator> propagator(double charge);
```

Comment:
constness does not look right at first sight.
I assume that this method is just here to return a pointer to the right propagator object depending on the charge. 
the user should not be able to modify this propagator object. 
The simulator object should also not be modified by the use of this method.

```c++
std::shared_ptr<const Propagator> propagator(double charge) const; 
```